WebAPI/Controllers/CategoriesController.cs

```diff
@@ -22,4 +22,25 @@ public async Task<ActionResult<List<CategoryDetailDto>>> GetAll()
 
         return Ok(result);
     }
+
+    [HttpGet]
```

Comment:
It's better to specify entity attribute your are searching by, in your case it is `id`. So, route should be represented as :
```suggestion
    [HttpGet("{id}:guid")]
```

Where `:guid` - constraint for use Guid type only

Same case for `ProductsController`